daemon/containerd/image_delete.go

```diff
@@ -45,18 +45,27 @@ import (
 // TODO(thaJeztah): implement ImageDelete "force" options; see https://github.com/moby/moby/issues/43850
 // TODO(thaJeztah): implement ImageDelete "prune" options; see https://github.com/moby/moby/issues/43849
 // TODO(thaJeztah): add support for image delete using image (short)ID; see https://github.com/moby/moby/issues/43854
-// TODO(thaJeztah): mage delete should send image "untag" events and prometheus counters; see https://github.com/moby/moby/issues/43855
+// TODO(thaJeztah): image delete should send prometheus counters; see https://github.com/moby/moby/issues/45268
 func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
 	parsedRef, err := reference.ParseNormalizedNamed(imageRef)
 	if err != nil {
 		return nil, err
 	}
 	ref := reference.TagNameOnly(parsedRef)
 
+	desc, err := i.resolveDescriptor(ctx, imageRef)
+	if err != nil {
+		return nil, err
+	}
+
+	imgID := string(desc.Digest)
 	err = i.client.ImageService().Delete(ctx, ref.String(), images.SynchronousDelete())
 	if err != nil {
 		return nil, err
 	}
 
+	i.LogImageEvent(imgID, imgID, "untag")
+	i.LogImageEvent(imgID, imgID, "delete")
```

Comment:
Actually - this might not be fully correct. The image is untagged, but the underlying content can still be around if there's other image that points to the same target (or there's other lease on that content).
On the other hand, we don't really control when that content gets deleted so if we don't do it at this point, we might never actually be able to send it (because the content can be GC'd directly by containerd later).